### ISSUE TITLE
Mailqueue robustness

### ISSUE THREAD

**Issue by Vekseid** [elkarte/Elkarte#2445]
Currently deleteMailQueueItems gets run on all ids in mail.subs.php regardless of the response gotten for said attempts.

We want to try to be a bit more intelligent, and only delete those ids that have been successfully queued on the mailserver.

Something else I'll try to take care of for 1.1

**Comment by emanuele45** [comment]
If I got it right, the entries are indeed deleted, but if the mailserver fails (or at least, if php is notified the sending failed) the entry is added back.

**Comment by Vekseid** [comment]
Yeah it looks like the issue is sometimes that doesn't work properly. Hrm.

**Comment by Vekseid** [comment]
This might be something that we can only really setup a better logging framework for so E/BMR can test when moving to Elkarte.
